core/issuers.js

```diff
@@ -116,28 +124,32 @@ Description must be truthy: ${description}`;
       return issuer;
     },
     destroyAll() {
-      mintController.destroyAll();
+      purseController.destroyAll();
+      paymentController.destroyAll();
     },
     destroy(amount) {
       amount = assay.coerce(amount);
       // for non-fungible tokens that are unique, destroy them by removing them from
       // the purses/payments that they live in
-      mintController.destroy(amount);
+      destroy(amount);
     },
     revoke(amount) {
       this.destroy(amount);
       return mint(amount);
     },
-    mint(initialBalance, _name = 'a purse') {
+    mint(initialBalance, unsafeName = 'a purse') {
```

Comment:
We're normally suspicious of incoming parameters. Sometimes we make this explicit in the parameter names, like `allegedFoo`, to really draw our vigilance to examining how we use it. For `unsafeName`, you coerce it to a string immediately. Afterwards there's nothing more to worry about. I'd just name the parameter `name` and turn the `const name` declaration below into an assignment.

Reply:
I'm usually skeptical of re-using variable names like this. Is there a reason it's safer in javascript than in other languages?

I think the usual guideline is that it's bad practice to change the value assigned to a variable so readers who skim the code might miss the change. I strongly prefer using a name like unsafeFoo, and declaring a new variable to hold the cleaned-up value.

Reply:
I agree with Chris in that I'd prefer not to re-assign to a variable.

Reply:
What about line 51 below?

Almost always in defensive programming, all arguments are things we're suspicious of, whether we call this unsafe* or alleged* notReliedOn* or something else. I am certainly in favor of any convention that helps us remember to be suspicious of these! However, it needs to be a convention we can apply consistently. 

With `alleged*` I chose only things for which we must remain suspicious event after the beginning of the function tries to normalize and validate. This, initialBalance on line 141 is not `allegedInitialBalance` or `unsafeInitialBalance`.

By what principle is `name` worth prefix but `initialBalance` not?

Regarding declaring new vs reassigning to old, I'm calm about it. We need a convention. But fixing line 142 and not 141 is jarring. ***If*** we don't come up with a naming convention for labeling all parameters we can't rely on, and ***if*** we continue to use unqualified names for most of these, then I would prefer reassignment for normal-named parameter variables that are immediately normalized to relied-upon values on entry to the function, before it does anything else. Otherwise, I agree that declaring new variables is likely better.

We should find principles to create conventions we can widely adhere to.

Reply:
Sounds good to me. I can create an issue for this to push the discussion out of this PR.